lib/src/css_box_widget.dart

```diff
@@ -59,8 +59,24 @@ class CssBoxWidget extends StatelessWidget {
     final direction = _checkTextDirection(context, textDirection);
     final padding = style.padding?.resolve(direction);
 
+    Width? maxWidthCalculated;
+    if (style.maxWidth != null && style.width != null) {
+      if (style.maxWidth!.unit == Unit.percent &&
+          style.width!.unit == Unit.px) {
+        // If our max is a percentage, we want to look at the size available and not be bigger than that.
+        try {
+          double width =
+              MediaQuery.of(context).size.width * (style.maxWidth!.value / 100);
+          maxWidthCalculated = Width(width, style.width!.unit);
+        } catch (_) {}
```

Comment:
silent errors are generally not a good idea - you should print the exception with a stacktrace at the very least (if this code happens to fail, you won't know unless you debug carefully the whole flow)